### PR TITLE
Forcefully disable versionlock plugin in DNF

### PR DESCRIPTION
**Pull request by ignatenkobrain**
```
[ INFO    ]: 09:39:08 | Uninstalling system packages (chroot)
[ INFO    ]: 09:39:08 | Using package manager backend: dnf
[ INFO    ]: 09:39:08 | --> package: linux-firmware
[ INFO    ]: 09:39:08 | Uninstall system packages (chroot)
[ DEBUG   ]: 09:39:08 | EXEC: [chroot /tmp/myimage8/build/image-root dnf --config /kiwi_v708wllp -y --releasever=8 autoremove linux-firmware]
[ INFO    ]: Processing: [########################################] 100%
[ ERROR   ]: 09:39:08 | KiwiPackagesDeletePhaseFailed: System package deletion failed: Package deletion failed: Error: Locklist not set
```

The versionlock plugin does not bring any benefit anyways as we
completely override plugin configs.

Also refactor configparser related stuff a bit.

Signed-off-by: Igor Raits <igor.raits@gmail.com>

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
@Conan-Kudo any specific test you are aware of?

**Comment by Conan-Kudo** [comment]
https://github.com/OSInside/kiwi/blob/master/test/unit/repository/dnf_test.py

**Comment by ignatenkobrain** [comment]
Oh, I added there the versionlock.conf but I'm not really sure what kinds of tests to put in there more.

**Comment by schaefi** [comment]
@ignatenkobrain I let the test pipeline start and you can see it fails on the tests:

* https://github.com/OSInside/kiwi/runs/6992620255?check_suite_focus=true

actually expected according to the changes you made. The changes looks good to me but we need to adapt the tests to follow the change. Let us know if you like us to assist with the tests

**Comment by schaefi** [comment]
@ignatenkobrain I added you as collaborator with write permissions to the project (pending invite) and hope you accept it ;) This allows you to create PRs from branches which automatically triggers all the github actions we have setup for kiwi. When people creating PRs from forks some of us has to approve that pipeline run manually which is understandable but also cumbersome if you know who is coding.

Thanks much for your effort :+1:

**Comment by Conan-Kudo** [comment]
> @ignatenkobrain I added you as collaborator with write permissions to the project (pending invite) and hope you accept it ;) This allows you to create PRs from branches which automatically triggers all the github actions we have setup for kiwi. When people creating PRs from forks some of us has to approve that pipeline run manually which is understandable but also cumbersome if you know who is coding.
> 
> Thanks much for your effort 👍

FWIW, PR pipelines for him will now start working automatically in this branch on his fork now that you did that.

**Comment by ignatenkobrain** [comment]
@schaefi actually I would appreciate help with these tests… The code works but the mocking somehow does not like it.

**Comment by schaefi** [comment]
@ignatenkobrain ok, I pushed a commit to fix the tests. If all goes well we should get a green check mark here soon

@Conan-Kudo can you review the changes with regards to dnf again ? I'm not sure about the changes, to me it looks good

Thanks

File: test/unit/repository/dnf_test.py [hunk 1]
```diff
@@ -33,21 +33,21 @@ def setup(self, mock_path, mock_config, mock_temp):
...
+                'pluginconfpath': '/shared-dir/dnf/pluginconf',
+                'keepcache': '1',
+                'debuglevel': '2',
+                'best': '1',
+                'obsoletes': '1',
+                'plugins': '1',
+                'gpgcheck': '0'
+            }),
+            call('main', {'enabled': '1'}),
+            call('main', {'enabled': '0'})
```
Comment (Conan-Kudo) [inline test/unit/repository/dnf_test.py:50]: This doesn't seem right? I assume this is supposed to be the plugin config?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: test/unit/repository/dnf_test.py (hunk 1 above)
Comment (ignatenkobrain) [reply test/unit/repository/dnf_test.py:50]: I'd say this part just checks what has been written to files. so it hightly depends on the order. And it was like this even before.

**Pull request merged**